src/Simplify.elm

```diff
@@ -7708,22 +7782,31 @@ arrayDetermineSize resources expressionNode =
             case AstHelpers.getSpecificFunctionCall ( [ "Array" ], "repeat" ) resources.lookupTable expressionNode of
                 Just repeatCall ->
                     Evaluate.getInt resources repeatCall.firstArg
-                        |> Maybe.map Exactly
+                        |> Maybe.andThen ignoreCollectionSizeIfEmpty
 
                 Nothing ->
                     Nothing
         , \() ->
             case AstHelpers.getSpecificFunctionCall ( [ "Array" ], "initialize" ) resources.lookupTable expressionNode of
                 Just repeatCall ->
                     Evaluate.getInt resources repeatCall.firstArg
-                        |> Maybe.map Exactly
+                        |> Maybe.andThen ignoreCollectionSizeIfEmpty
 
                 Nothing ->
                     Nothing
         ]
         ()
 
 
+ignoreCollectionSizeIfEmpty : Int -> Maybe CollectionSize
+ignoreCollectionSizeIfEmpty n =
+    if n <= 0 then
+        Nothing
```

Comment:
Why not return `Exactly 0`? We know the resulting size will be zero.

Reply:
Because then we get 2 errors reported for the same issue (see https://github.com/jfmengels/elm-review-simplify/pull/191#discussion_r1335041738)
We get one error for `Array.repeat -2 x` and we get one for `Array.repeat (Array.repeat -2 x)`, which is not a great experience. Happy to be convinced otherwise though.

Reply:
Aren't we doing that for multiple errors already?
Simple example being `Set.size (Set.fromList [])` where we both want to simplify `Set.fromList []` to `Set.empty` and `Set.size ...` to `0`.

Reply:
Alright. I'll change it :+1: